url8867e6a4e2_cyberpython_JAwesomeChart_tgz-pJ8-com_gmigdos_jawesomechart_renderers_twodimensional_horizontal_ColumnChartRendererJ8/src/com/gmigdos/jawesomechart/core/Legend.java

```diff
@@ -0,0 +1,313 @@
+                if (entryWidth > maxLineWidth) {
+                    context.adjustFontSizeToFitTextInWidth("   " + longestName, width - 2 * paddingHorizontal);
+                    markerDimension = context.getStandardLineHeight();
+                }
+                String name;
+                boolean first = true;
+                for (Iterator<DataSeries> it = series.iterator(); it.hasNext();) {
+                    DataSeries dataSeries = it.next();
+                    name = dataSeries.getName();
+                    if (name != null) {
+                        if(first){
+                            first = false;
+                            totalHeight += context.getLineHeight(font, name)+verticalGap;
+                        }
+                        entryWidth = markerDimension + horizontalGap + context.calculateStringWidth(name);
+                        currentWidthSum += entryWidth;
+                        currentLineMaxHeight = Math.max(context.getLineHeight(font, name), currentLineMaxHeight);
+                        if (currentWidthSum > maxLineWidth) {
+                            totalHeight += currentLineMaxHeight;
+                            if(it.hasNext()){
+                                totalHeight += verticalGap;
+                            }
+                            currentWidthSum = entryWidth;
+                        }
+                        currentWidthSum += 4*horizontalGap;
+                        
+                    }
+                }
+                font = context.getFont();
+                totalHeight += paddingVertical;
+                height = totalHeight;
+            }
+        }
+    }
+
+    public double getHeight() {
+        return height;
+    }
+
+    public double getWidth() {
+        if(isPlacedOnLeftOrRight()){
+            return leftOrRightWidth;
+        }
+        return topOrBottomWidth;
+    }
+
+    public void setLeftOrRightWidth(double leftOrRightWidth) {
+        this.leftOrRightWidth = leftOrRightWidth;
+        updateSize();
+    }
+
+    public void setTopOrBottomWidth(double topOrBottomWidth) {
+        this.topOrBottomWidth = topOrBottomWidth;
+        updateSize();
+    }
+
+    public DataSeriesList getSeries() {
+        return series;
+    }
+
+    public void setSeries(DataSeriesList series) {
+        this.series = series;
+    }
+
+    public double getVerticalGap() {
+        return verticalGap;
+    }
+
+    public void setVerticalGap(double verticalGap) {
+        this.verticalGap = verticalGap;
+    }
+
+    public double getHorizontalGap() {
+        return horizontalGap;
+    }
+
+    public void setHorizontalGap(double horizontalGap) {
+        this.horizontalGap = horizontalGap;
+    }
+
+    public Font getFont() {
+        return font;
+    }
+
+    public void setFont(Font font) {
+        this.font = font;
+    }
+
+    public void draw() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 49 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.